src/utils.ts

```diff
@@ -55,6 +56,12 @@ export const fetchAllChainMetadataMap = async () => {
   const allMetadata: ChainMap = {};
   for (const metadata_i of resp.data as EthListChainMetadata[]) {
     console.log("chain ", metadata_i.chainId);
+    const viemChain = viemChainsMap[metadata_i.chainId];
+
+    if (viemChain) {
+      metadata_i.rpc = [...viemChain.rpcUrls.default.http, ...metadata_i.rpc]
```

Comment:
wondering if we should just supply the viem chain rpc's only if they exist since the ethereum-list ones are so flaky

Reply:
I don't the non viem rpcs are necessarily incorrect, so I don't see any reason to not include them

One thing we can do is filter out the rpcs that need API keys like `https://mainnet.infura.io/v3/${INFURA_API_KEY}`, which is present in a couple of chains

Reply:
sounds good. I'm okay with including both. as long as we know to start from the front of the array.

we have options to pass the infura or alchemy api get in the `getChainMetadata` params and it will replace those api keys and return a useable RPC url